app/domain/authentication/authn_gcp/authenticator.rb

```diff
@@ -5,27 +5,37 @@ module AuthnGcp
 
     Authenticator = CommandClass.new(
       dependencies: {
-        validate_resource_restrictions: ValidateResourceRestrictions.new,
+        validate_resource_restrictions: Authentication::Common::ValidateResourceRestrictions.new,
+        authentication_request_class:   AuthenticationRequest,
         logger:                         Rails.logger
       },
       inputs:       [:authenticator_input]
     ) do
 
       extend Forwardable
-      def_delegators :@authenticator_input, :authenticator_name, :account, :username, :credentials
+      def_delegators :@authenticator_input, :authenticator_name, :service_id, :account, :username, :credentials
 
       def call
-        validate_resource_restrictions
+        authentication_request = create_authentication_request
+        validate_resource_restrictions(authentication_request)
       end
 
       private
 
-      def validate_resource_restrictions
+      def create_authentication_request
```

Comment:
the method name should be `authentication_request` to indicate the output.

Reply:
But this is not an accessor, it wraps an instance creation.
What's wrong with this name? to me it looks very intuitive.

Reply:
if a method returns an object (as this method does) then we give it a name that indicates the returned object. in this case it's `authentication_request`. 

Here i would have both methods `create_authentication_request` and `authentication_request` in the manner i suggested here: https://github.com/cyberark/conjur/pull/1918/files/58a5b8b4db11d7c66cc786020fa76332d823201e#r515952095

Reply:
Although I prefer the explicit way, I understand this style, and adhere to it.
Done.

Reply:
thanks you very much, i appreciate it!